packages/mui-system/src/useThemeProps/useThemeProps.js

```diff
@@ -7,5 +8,5 @@ export default function useThemeProps({ props, name, defaultTheme, themeId }) {
   if (themeId) {
     theme = theme[themeId] || theme;
   }
-  return getThemeProps({ theme, name, props });
+  return React.useMemo(() => getThemeProps({ theme, name, props }), [theme, name, props]);
```

Comment:
I doubt that `props` will always be a new object on every render?
see this sandbox https://codesandbox.io/p/sandbox/react-new

Reply:
> React.useMemo would not help.

Memoizing the return value of `getThemeProps` does fix this specific issue we're seeing, see https://github.com/mui/mui-x/pull/17490